src/omlt/linear_tree/lt_definition.py

```diff
@@ -0,0 +1,408 @@
+        The modified leaves dict without any bounds that are listed as None
+    """
+    leaf_indices = np.array(list(leaves.keys()))
+    leaf_one = leaf_indices[0]
+    features = np.arange(0, len(leaves[leaf_one]["slope"]))
+
+    for leaf in leaf_indices:
+        for feat in features:
+            if leaves[leaf]["bounds"][feat][0] is None:
+                leaves[leaf]["bounds"][feat][0] = input_bounds[feat][0]
+            if leaves[leaf]["bounds"][feat][1] is None:
+                leaves[leaf]["bounds"][feat][1] = input_bounds[feat][1]
+
+    return leaves
+
+
+def _parse_tree_data(model, input_bounds):
+    """
+    This function creates the data structures with the information required
+    for creation of the variables, sets, and constraints in the pyomo
+    reformulation of the linear model decision trees. Note that these data
+    structures are attributes of the LinearTreeDefinition Class.
+
+    Arguments:
+        model -- Trained linear-tree model or dic containing linear-tree model
+            summary (e.g. dict = model.summary())
+
+    Returns:
+        leaves - Dict containing the following information for each leaf:
+            1) 'slope' - The slope of the fitted line at that leaf
+            2) 'intercept' - The intercept of the line at that lead
+            3) 'parent' - The parent split or node of that leaf
+        splits - Dict containing the following information for each split:
+            1) 'children' - The child nodes of that split
+            2) 'col' - The variable(feature) to split on (beginning at 0)
+            3) 'left_leaves' - All the leaves to the left of that split
+            4) 'right_leaves' - All the leaves to the right of that split
+            5) 'parent' - The parent node of the split. Node zero has no parent
+            6) 'th' - The threshold of the split
+            7) 'y_index' - Indices corresponding to Mistry et. al. y binary
+                    variable
+        vars_dict - Dict of tree inputs and their respective thresholds
+
+    Raises:
+            Exception: If input dict is not equal to model.summary()
+            Exception: If input model is not a dict or linear-tree instance
+    """
+    # Create the initial leaves and splits dictionaries depending on the
+    # instance of the model (can be either a LinearTreeRegressor or dict).
+    # Include checks to ensure that the input dict is the model summary which
+    # is obtained by calling the summary() method contained within the
+    # linear-tree package (e.g. dict = model.summary())
+    if isinstance(model, lineartree.lineartree.LinearTreeRegressor) is True:
+        leaves = model.summary(only_leaves=True)
+        splits = model.summary()
+    elif isinstance(model, dict) is True:
+        splits = model
+        leaves = {}
+        num_splits_in_model = 0
+        count = 0
+        # Checks to ensure that the input nested dictionary contains the
+        # correct information
+        for entry in model:
```

Comment:
Something is strange in this code block below. (Maybe an indentation problem.). Please double-check.

Reply:
You are correct. I consolidated this into one else statement.